cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -120,16 +120,20 @@ macro (build_addon target prefix libs)
             foreach(src_file ${USED_SOURCES})
               file(STRINGS ${src_file} BIN_ADDON_SRC_PARTS)
               foreach(loop_var ${BIN_ADDON_SRC_PARTS})
+                string(REGEX REPLACE ".*#include" "#include" loop_var "${loop_var}")
                 string(FIND "${loop_var}" "#include" matchres)
                 if("${matchres}" EQUAL 0)
                   string(REPLACE " " ";" loop_var "${loop_var}")
                   list(GET loop_var 1 include_name)
                   string(REGEX REPLACE "[<>\"]|kodi/" "" include_name "${include_name}")
                   if(include_name MATCHES ${depend_header})
                     set(ADDON_DEPENDS "${ADDON_DEPENDS}\n<import addon=\"${${xml_entry_name}}\" version=\"${${depends_name}}\"/>")
-                    # Inform with them the addon header about used type
-                    add_definitions(-D${used_type_name})
-                    message(STATUS "Added usage definition: ${used_type_name}")
+                    # Inform with them the addon header about used type, if not present before
+                    get_directory_property(CURRENT_DEFS COMPILE_DEFINITIONS)
+                    if(NOT CURRENT_DEFS MATCHES "${used_type_name}")
+                      add_definitions(-D${used_type_name})
```

Comment:
What is that needed for?

Reply:
The `add_definitions` is needed for `#if` here: https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h#L178-L256.

Where addon returns `"0.0.0"` if not needed and becomes ignored from Kodi.

Reply:
Only greped for `ADDON_GLOBAL_VERSION_MAIN_USED` and that didn't show anything.